apps/auth/models.py

```diff
@@ -31,3 +32,16 @@ def __init__(self, username: str, email: str, password=None):
 
     def set_password(self, password):
         self.password = generate_password_hash(password)
+
+    def get_last_10_quiz_attempts(self):
+        this_user_attempts = UserQuizAttempts.query.filter_by(user_id=self.id).order_by(UserQuizAttempts.date.desc()).limit(10).all()
+        if this_user_attempts == []:
+            return "You don't have any quiz attempts yet"
+        else:
+            final_table = []
```

Comment:
This is not necessary, you may simply return a query result

Reply:
If I just return the query result, it returns it in a form that can't be displayed, it must be altered somewhere, why not here?

Reply:
If i return just the query result it is in a form, that has to be altered to by displayed on site, why not alter it here?

Reply:
Why query result can not be displayed?
The example:
[LINK](https://blog.miguelgrinberg.com/post/the-flask-mega-tutorial-part-ix-pagination)
Query result may be treated as standard iterable, I'll check this on this code, today's evening

Reply:
If you just return the query, you have something like that:
![image](https://user-images.githubusercontent.com/93865526/151039703-581481d5-8d14-4f9a-ba39-1ede366f0298.png)
Which is not the information we want to have displayed, that's why there is the whole final table thing

Reply:
This is a QueryList of QuizAttempts objects, after iterting throuugh those, you may acces each param like this:
```
{% for attempt in attempts %}
{{attempt.score}}
{% endfor %}

```